Training_node/first_project/app/middlewares/auth.js

```diff
@@ -1,16 +1,26 @@
 const sessionManager = require('./../services/sessionManager'),
-  User = require('../models').user;
+  User = require('../models').user,
+  logger = require('../logger');
 
-exports.secure = (req, res, next) => {
-  const auth = req.headers[sessionManager.HEADER_NAME];
-
-  if (auth) {
-    const user = sessionManager.decode(auth);
+const auth = (req, res, next, isAdmin = false) => {
+  const authHeader = req.headers[sessionManager.HEADER_NAME];
+  if (authHeader) {
+    const user = sessionManager.decode(authHeader);
     return User.getByEmail(user)
       .then(u => {
         if (u) {
-          req.user = u;
-          next();
+          if (isAdmin) {
```

Comment:
why this two ifs?

Reply:
It's similar create(createAdmin and createRegular) but in this context. If you want verify the log-in user regular, if the exist there is enough but if you want verify the log-in admin also verify if the user have permissions.